src/liquid/components/ld-notification/ld-notification.tsx

```diff
@@ -0,0 +1,190 @@
+  type: 'info' | 'warn' | 'alert'
+  content: string
+  timeout?: number
+}
+
+const DEFAULT_NOTIFICATION_TIMEOUT = 6000
+const FADE_TRANSITION_DURATION = 200
+
+@Component({
+  tag: 'ld-notification',
+  styleUrl: 'ld-notification.css',
+  shadow: false,
+})
+export class LdNotification {
+  /**
+   * Notification placement within the screen.
+   */
+  @Prop() placement: 'top' | 'bottom' = 'top'
+
+  @State() queue: Notification[] = []
+  @State() queueDismissed: Notification[] = []
+
+  @State() dismissTimeout: NodeJS.Timeout | null
+
+  @State() fadeoutTimeouts: (NodeJS.Timeout | null)[] = []
+
+  @State() currentNotification?: Notification
+
+  @Watch('currentNotification')
+  updateDismissTimeout() {
+    clearTimeout(this.dismissTimeout)
+
+    if (!this.currentNotification) return
+    if (this.currentNotification.type === 'alert') return
+    if (this.currentNotification.timeout === 0) return
+
+    this.dismissTimeout = setTimeout(() => {
+      this.handleNotificationDismiss()
+    }, this.currentNotification.timeout || DEFAULT_NOTIFICATION_TIMEOUT)
+  }
+
+  @Listen('ldNotification', {
+    target: 'window',
+    passive: true,
+  })
+  handleNotification(ev: CustomEvent<Notification>) {
+    ev.stopImmediatePropagation()
+    const newNotification = ev.detail
+
+    // If the same notification is already in queue (same content, same type), ignore this notification.
+    const inQueue = this.queue.find(
+      (notification) =>
+        notification.content === newNotification.content &&
+        notification.type === newNotification.type
+    )
+    if (inQueue) return
+
+    // Insert by relevance, whith error notifications being more relevant than non-error notifications.
+    if (newNotification.type === 'alert') {
+      this.queue = [...this.queue, newNotification]
+      this.currentNotification = newNotification
+      return
+    }
+
+    const firstErrorNotificationIndex = this.queue.findIndex(
+      (notification) => notification.type === 'alert'
+    )
+    if (firstErrorNotificationIndex === -1) {
+      this.queue = [...this.queue, newNotification]
+      this.currentNotification = newNotification
+      return
+    }
+
+    this.queue.splice(firstErrorNotificationIndex, 0, newNotification)
+    this.queue = [...this.queue]
+  }
+
+  @Listen('ldNotificationDismiss', {
+    target: 'window',
+    passive: true,
+  })
+  handleNotificationDismiss() {
+    if (!this.currentNotification) return
+
+    this.queueDismissed.unshift(this.queue.pop())
+    this.queueDismissed = [...this.queueDismissed]
+    this.queue = [...this.queue]
+    this.currentNotification = this.queue[this.queue.length - 1]
+
+    this.fadeoutTimeouts.push(
+      setTimeout(() => {
+        this.queueDismissed = this.queueDismissed.slice(0, -1)
+      }, FADE_TRANSITION_DURATION)
+    )
+  }
+
+  @Listen('ldNotificationClear', {
+    target: 'window',
+    passive: true,
+  })
+  handleNotificationClear() {
+    this.queueDismissed = [...this.queue]
```

Comment:
I don't want to do that. I want a copy. Otherwise if I change the content of this.queue, the content of this.queueDismissed will change as well, since both point to the same object,